tests/async_engine/test_request_tracker.py

```diff
@@ -47,8 +47,10 @@ async def test_request_tracker():
     assert tracker.new_requests_event.is_set()
     await tracker.wait_for_new_requests()
     new, aborted = tracker.get_new_and_aborted_requests()
-    assert len(aborted) == 1
-    assert "4" in aborted
+    # aborted new requests will cancel each other out -
+    # there's no need for them to propagate into the
+    # engine
+    assert not aborted
```

Comment:
Could you explain why this is desired behavior?

Reply:
Sorry, I must have done a bad job of this in the comment :)

This method is used in the main async engine model loop to get new and cancelled requests since the last iteration, it will begin processing the new ones and abort the processing of cancelled ones. The case here is when a new request is aborted before this method is called - it should be like it never happened from the model loop's pov.

Reply:
Ah I understand now, thanks for the explanation!